chain/chain/src/runtime/mod.rs

```diff
@@ -567,20 +567,20 @@ impl RuntimeAdapter for NightshadeRuntime {
         shard_layout: &ShardLayout,
         gas_price: Balance,
         state_root: StateRoot,
-        transaction: &SignedTransaction,
+        tx: &ValidatedTransaction,
         current_protocol_version: ProtocolVersion,
     ) -> Result<(), InvalidTxError> {
         let runtime_config = self.runtime_config_store.get_config(current_protocol_version);
 
         let cost =
-            tx_cost(runtime_config, &transaction.transaction, gas_price, current_protocol_version)?;
-        let shard_uid = shard_layout.account_id_to_shard_uid(transaction.transaction.signer_id());
+            tx_cost(runtime_config, &tx.get().transaction, gas_price, current_protocol_version)?;
+        let shard_uid = shard_layout.account_id_to_shard_uid(tx.get().transaction.signer_id());
```

Comment:
I'd explore either `impl Deref for ValidatedTransaction { type Target=SignedTransaction; ... }` (and similarly for `SignedTransaction` → `Transaction`) or implementing accessors like these on `ValidatedTransaction` to avoid going back to `SignedTransaction` manually everywhere.

Reply:
I am less keen on using `Deref` as I feel like without proper IDE support, it might make it a bit challenging for readers to figure out which types are being coerced into.  I introduced some more accessors functions and made their names a bit smaller.